### DISCUSSION TITLE
More and configurable keyboard bindings

### DISCUSSION THREAD

**Discussion by dov** [xournalpp/xournalpp#3686] [Ideas]
**Is your feature request related to a problem? Please describe.**

I am currently using krita for writing notes and as a Whiteboard. However in Krita I'm missing several options that XournalPP has, and I therefore tried it out (again). However there are several things in XournalPP that irritate me. I should probably open a different issue about each one, but I'll leave this note here as a a probe. :-)

Here are some behaviors that I suggest that XournalPP import from Krita:

1. Control-A to select all objects on the screen. Nice for quickly erasing everything when working as a white board.
2. "e" (or another configurable keybord binding) to swap between eraser and pen
3. Shift-stylus drag to change the pen size. 
4. In the quick menu (left margin), replace the two pen sizes (optionaly) with a slider.
4. Make brush size (optionally) seen on screen in points.
5. Option "[" and "]" (or something other configurable) for changing the pen size.
6. Allow a quick palette like krita with recent colors, brush options, and other quick options. This is great when working in full screen.
7. Configurable keybindings for going between each tool. E.g. I like to use "B" brush. "E" ellipse. "R" Rectangle. Shift-R select rectangle, etc.
8. Provide the following three brushes and make it easy to change between them:
   1. Fixed size, fixed opacity (and allow changing the opacity)
   2. Fixed opacity, varying thickness based on pressure
   3. Fixed thickness, varying opacity based on pressure.

Sorry for the ramble. If there is a better forum for general kind of discussions, please let me know.

**Comment by rolandlo** [comment]
Thanks for your suggestions. Let me give a partial answer to them.
- 1), is an interesting suggestion. This is currently not supported, but it would be easy to implement.
- 2), 3), 6) and 8) can be achieved (with little effort) via a [plugin](https://xournalpp.github.io/guide/plugins/plugins/)
- 7), 9) could potentially also be achieved via plugins, although more work is required for that. Also note that for 7) there is a floating toolbox, which may partially solve your problem 

I will move this issue to Github discussions, as that place looks more appropriate.

**Reply by dov** [reply]
Thanks a lot for the reply! I'll certainly check out the plugin options.

Let me also use this discussion board to make a shameless plug for a plugin that I made for Krita that is somehow overlapping with functionality provided by xournalpp:

https://github.com/dov/krisenter
